### PR TITLE
[CANNOLI-33] Use ADAM tab5 formatter for bowtie

### LINKED ISSUE #33: Use ADAM tab5 formatter for bowtie

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #33 
Depends on https://github.com/bigdatagenomics/adam/pull/1522

No reads are processed however
```
$ ADAM_MAIN=org.bdgenomics.cannoli.Cannoli \
  adam-submit \
  --jars target/cannoli-spark2_2.11-0.1-SNAPSHOT.jar \
  -- \
  bowtie \
  -bowtie_index GCA_000001405.25_GRCh38.p10_genomic \
  -print_metrics -single \
  NIST7035_TAAGGCGA_L002.ifq \
  NIST7035_TAAGGCGA_L002.sam

Using ADAM_MAIN=org.bdgenomics.cannoli.Cannoli
Using SPARK_SUBMIT=/usr/local/bin/spark-submit
...
# reads processed: 0
# reads with at least one reported alignment: 0 (0.00%)
# reads that failed to align: 0 (0.00%)
No alignments
# reads processed: 0
# reads with at least one reported alignment: 0 (0.00%)
# reads that failed to align: 0 (0.00%)
No alignments
```

Unless I'm doing something wrong, `--12 -` doesn't seem to work
```
$ cat fragments.tab5 
H06HDADXX130110:2:2116:3345:91806	GTTAGGGTTAGGGTTGGGTTAGGGTTAGGGTTAGGGTTAGGGGTAGGGTTAGGGTTAGGGGTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGGTAGGGCTAGGGTTAAGGGTAGGGTTAGCGAAAGGGCTGGGGTTAGGGGTGCGGGTACGCGTAGCATTAGGGCTAGAAGTAGGATCTGCAGTGCCTGACCGCGTCTGCGCGGCGACTGCCCAAAGCCTGGGGCCGACTCCAGGCTGAAGCTCAT	>=<=???>?>???=??>>8<?>=2=<===1194<?;:?>>?#3==>###########################################################################################################################################################################################################	TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTACCCCTAACCCTAACCCTAACCCTAACCCGTACCCTAAACCCAACCCTAACCACAAAGCAAATCCCAACCTTAACCGGAACCCGAAATCTCGCAGCAAATCTGCAGTAGAGACGCAGACTCAACCATGCGTCTATTAGTACGCATTATCATTGCCTCATGCTTCTTAAGTACAGAGAGATGAC	==;<?>@@@<>>@??<>>???<=>>?>:><@?4=:>7=5=>:=@;'@A?########################################################################################################################################################################################################
...

$ bowtie -S GCA_000001405.25_GRCh38.p10_genomic --12 fragments.tab5 
@HD	VN:1.0	SO:unsorted
@SQ	SN:CM000663.2	LN:248956422
...
@PG	ID:Bowtie	VN:1.2	CL:"bowtie-align --wrapper basic-0 -S GCA_000001405.25_GRCh38.p10_genomic --12 fragments.tab5"
H06HDADXX130110:2:2116:3345:91806	77	*	0	0	*	*	0	0	GTTAGGGTTAGGGTTGGGTTAGGGTTAGGGTTAGGGTTAGGGGTAGGGTTAGGGTTAGGGGTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGGTAGGGCTAGGGTTAAGGGTAGGGTTAGCGAAAGGGCTGGGGTTAGGGGTGCGGGTACGCGTAGCATTAGGGCTAGAAGTAGGATCTGCAGTGCCTGACCGCGTCTGCGCGGCGACTGCCCAAAGCCTGGGGCCGACTCCAGGCTGAAGCTCAT	>=<=???>?>???=??>>8<?><=2=<===1194<?;:?>>?#3==>###########################################################################################################################################################################################################	XM:i:0
H06HDADXX130110:2:2116:3345:91806	141	*	0	0	*	*	0	0	TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTACCCCTAACCCTAACCCTAACCCTAACCCGTACCCTAAACCCAACCCTAACCACAAAGCAAATCCCAACCTTAACCGGAACCCGAAATCTCGCAGCAAATCTGCAGTAGAGACGCAGACTCAACCATGCGTCTATTAGTACGCATTATCATTGCCTCATGCTTCTTAAGTACAGAGAGATGAC	==;<?>@@@<>>@??<>>???<=>>?>:><@?4=:>7=5=>:<=@;'@A?########################################################################################################################################################################################################	XM:i:0
...

$ cat fragments.tab5 | bowtie -S GCA_000001405.25_GRCh38.p10_genomic --12 -
@HD	VN:1.0	SO:unsorted
@SQ	SN:CM000663.2	LN:248956422
...
@PG	ID:Bowtie	VN:1.2	CL:"bowtie-align --wrapper basic-0 -S GCA_000001405.25_GRCh38.p10_genomic --12 -"
# reads processed: 0
# reads with at least one reported alignment: 0 (0.00%)
# reads that failed to align: 0 (0.00%)
No alignments
```

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, retest this please

File: scripts/publish_scaladoc.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+#!/bin/bash
```
Comment (devin-petersohn) [inline scripts/publish_scaladoc.sh]: Is this the extent of the script?

File: src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala [hunk 2]
```diff
@@ -31,6 +31,7 @@ private[cannoli] object QuerynameGrouper extends Serializable {
   def apply(rdd: AlignmentRecordRDD): FragmentRDD = {
     new FragmentRDD(apply(rdd.rdd),
       rdd.sequences,
-      rdd.recordGroups)
+      rdd.recordGroups,
+      optPartitionMap = None)
```
Comment (devin-petersohn) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: We lose some information here about whether or not the data is sorted. This should be fixed in https://github.com/bigdatagenomics/adam/pull/1561. Under the new method, it does force a recompute on the `optPartitionMap`, but at least we would keep track of whether or not the data is sorted.

Comment (fnothaft) [inline src/main/scala/org/bdgenomics/cannoli/util/QuerynameGrouper.scala]: By definition, the data is _not_ coordinate sorted here.

File: scripts/publish_scaladoc.sh (hunk 1 above)
Comment (fnothaft) [inline scripts/publish_scaladoc.sh]: This is intentionally a stub for later.

**Comment by heuermh** [comment]
I'm not sure how, I've git-borked the commits on this one and can't seem to unsuck them.  The problem with bowtie itself has been fixed in version 1.2.1.1, and is available in homebrew-science.  I'd like to close this unmerged and try again with separate pull requests.

**Comment by fnothaft** [comment]
Closing as unmerged, @heuermh will reopen as separate PRs.

**Pull request closed without merging**